crates/node/src/lib.rs

```diff
@@ -64,7 +63,7 @@ pub struct RollupNodeManager<C, EC, P, L1P, L1MP, CS> {
     /// An indexer used to index data for the rollup node.
     indexer: Indexer<CS>,
     /// The consensus algorithm used by the rollup node.
-    consensus: C,
+    consensus: Box<dyn Consensus + Send>,
```

Comment:
Should we constrain the `Consensus` trait to be `Send + Debug` instead?

```rust
pub trait Consensus: std::fmt::Debug + Send {
    /// Validates a new block with the given signature.
    fn validate_new_block(
        &self,
        block: &ScrollBlock,
        signature: &Signature,
    ) -> Result<(), ConsensusError>;
}
```
```suggestion
    consensus: Box<dyn Consensus>,
```